openverse_catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -8,266 +8,173 @@
 
 Notes:                  https://www.europeana.eu/api/v2/search.json
 """
-
 import argparse
 import logging
 from datetime import datetime, timedelta, timezone
 
+import common
 from airflow.models import Variable
 from common.licenses import get_license_info
 from common.loader import provider_details as prov
-from common.requester import DelayedRequester
-from common.storage.image import ImageStore
-from requests.exceptions import JSONDecodeError
+from providers.provider_api_scripts.provider_data_ingester import ProviderDataIngester
 
 
-logging.basicConfig(
-    format="%(asctime)s - %(name)s - %(levelname)s:  %(message)s", level=logging.INFO
-)
 logger = logging.getLogger(__name__)
+logging.getLogger(common.urls.__name__).setLevel(logging.WARNING)
+
+
+class EuropeanaRecordBuilder:
+    """
+    A small class to contain the record building functionality
+    and simplify testing a bit.
+    """
+
+    def get_record_data(self, data: dict) -> dict:
+        record = {
+            "foreign_landing_url": self._get_foreign_landing_url(data),
+            "image_url": data.get("edmIsShownBy")[0],
+            "foreign_identifier": data.get("id"),
+            "meta_data": self._get_meta_data_dict(data),
+            "title": data.get("title")[0],
+            "license_info": get_license_info(
+                license_url=self._get_license_url(data.get("rights"))
+            ),
+        }
+
+        data_providers = set(record["meta_data"]["dataProvider"])
+        eligible_sub_providers = {
+            s
+            for s in EuropeanaDataIngester.sub_providers
+            if EuropeanaDataIngester.sub_providers[s] in data_providers
+        }
+        if len(eligible_sub_providers) > 1:
+            raise Exception(
+                f"More than one sub-provider identified for the "
+                f"image with foreign ID {record['foreign_identifier']}"
+            )
+
+        return record | {
+            "source": (
+                eligible_sub_providers.pop()
+                if len(eligible_sub_providers) == 1
+                else EuropeanaDataIngester.providers["image"]
+            )
+        }
+
+    def _get_license_url(self, license_field) -> str | None:
+        if len(license_field) > 1:
+            logger.warning("More than one license field found")
```

Comment:
Not sure if it should be done in this PR, but I think this log should be downgraded to `debug`